tracer/src/Datadog.Trace/AppSec/RateLimiterTimer.cs

```diff
@@ -0,0 +1,56 @@
+// <copyright file="RateLimiterTimer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.AppSec
+{
+    internal class RateLimiterTimer : IDisposable
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<RateLimiterTimer>();
+        private readonly Timer _timer;
+        private readonly int _traceRateLimit;
+        private int tracesCount;
+
+        public RateLimiterTimer(int traceRateLimit)
+        {
+            _traceRateLimit = traceRateLimit;
+            var now = DateTime.UtcNow;
+            var intwoSecs = DateTime.UtcNow.AddSeconds(2);
+            var onlyWithSeconds = new DateTime(intwoSecs.Year, intwoSecs.Month, intwoSecs.Day, intwoSecs.Hour, intwoSecs.Minute, intwoSecs.Second);
+            var diff = onlyWithSeconds.Add(-now.TimeOfDay);
+
+            _timer = new Timer(
+                   _ =>
+                   {
+                       Reset();
+                   },
+                   null,
+                   diff.TimeOfDay,
+                   TimeSpan.FromSeconds(1));
+        }
+
+        public int RateLimiterCounter => tracesCount;
+
+        public void Dispose() => _timer.Dispose();
+
+        public void Reset()
+        {
+            Interlocked.Exchange(ref tracesCount, 0);
+        }
+
+        /// <summary>
+        /// check if a trace can be added, otherwise increase the amount of exceeded traces
+        /// </summary>
+        /// <returns>returns the number of exceeded traces, 0 if ok</returns>
```

Comment:
Are these comments correct? It looks like the code always increments, and the return value is count - limit, which will be negative unless the limit is exceeded, if I understand correctly.

Reply:
Oh, I see. It returns the number of _exceeded_ traces (so anything above 0 means you're over the limit).